### ISSUE TITLE
How to find entities that have been soft deleted

### ISSUE THREAD

**Issue by baoduy** [JonPSmith/EfCore.GenericServices#10]
Hi @JonPSmith 
As the QueryFIlter is a feature allow to define a default filter for the query when reading from the database if useful when We are using soft delete mechanism. However when ReadSingle by entity Id it should call IgnoreQueryFilters as we want to load an identify item via the keys.

Shall We call IgnoreQueryFilters by default for ReadSingle? I could able to provide a pull request for this change?

Thank and Regards

**Comment by JonPSmith** [comment]
Hi @baoduy,

I had to think this through as I am working a client's application that uses soft delete. I rejected the idea of adding `IgnoreQueryFilters` to the ReadSingle because it could cause problems when someone is using query filters to implement a multi-tenant system.

I have thought about adding a `ReadWithPreQueryFilter` method (see [this example](https://github.com/JonPSmith/EfCore.GenericServices/blob/TryPreQuery/GenericServices/PublicButHidden/CrudServices.cs#L124#L129)). That would allow you to add a `IgnoreQueryFilters` on the front. I have held off because I want to consider some other changes.

In the meantime here is the code I use to find entries that have been soft deleted (in this case for a Web API return). It uses the DbContext directly and links into GenericServices's AutoMapper config. I hope this helps:
```csharp
[HttpGet("getdeleted")]
public async Task<ActionResult<WebApiMessageAndResult<List<TDisplayDto>>>> GetDeletedAsync()
{
    var deletedItems = await _context.Set<TEntity>().IgnoreQueryFilters().AsNoTracking()
        .Where(x => ... any multi-tenant filter ... && x.FilterFlag.HasFlag(FilterFlags.SoftDeleted))
        .ProjectTo<TDisplayDto>(_genericServicesConfig.MapperReadConfig).ToListAsync();
    var status = new StatusGenericHandler();
    return status.Response(deletedItems);
}
```
NOTE: The `TDisplayDto` type is one of GenericService styled DTOs, e.g. one that has the interface `ILinkToEntity<TEntity>`.

**Comment by JonPSmith** [comment]
I'm leaving this open as I may implement `ReadManyWithPreQueryNoTracked` sometime in the future.

**Comment by baoduy** [comment]
Hi @JonPSmith. I also was hardly thinking about this and agree with you that changes the ReadSingle behavior may cause many business issue instead of implementing a new method for that.

I may able to give a full request if you ok? Just confirm the method name is `ReadManyWithPreQueryNoTrack` or `ReadManyWithoutPreQueryNoTrack`?

Thanks and Regards

**Comment by JonPSmith** [comment]
Hi @baoduy,

Thank you for the offer, but I have found its very easy to add new commands, but you are stuck with them forever. I now delay a bit until I have a better handle on the user cases for the command. At the moment there is a fairly simple work around so I'm not going to add that feature at the moment.

So I suggest you either use the work around I showed you earlier or clone the repo and add the method for your local use.
